examples/solvers/pdhg_denoising_L1_HuberTV.py

```diff
@@ -0,0 +1,135 @@
+"""Total variation denoising using PDHG.
```

Comment:
Well, yes and no. While the application is almost the same, the choice of parameters for the algorithms are very different. The acceleration when parts of the objective are smooth is not widely used so I would like to keep this example.

Reply:
I see your point. But given that it's a lot of extra code just for that, how about having the alternative together with the parameter choice, commented out and next to the used definition? Then users can simply switch between the two by commenting in and out a couple of lines.

Reply:
To me there are too many changes the user would need to make. Changing the noise model, changing L2 to L1, changing the parameters, changing the input of PDHG, changing the plotting. Maybe we can combine this example with something else that ODL wants to showcase? I don't mind too much which data fit, prior or image we use as long as we have an example for the 1/k^2 case. Maybe something with colour and vectorial TV or nuclear norms?

Reply:
Another possibility is to make a big
```python
if data_term == 'L1':
    # Do the L1 case, including plotting and whatnot
elif data_term == 'L2':
    # Other case
```
Since it's a rather elaborate example anyway, making one large example from two slightly-less-than-large would still be good IMO.
What's your opinion @adler-j?

Reply:
At first glance, I really like your idea as it reduces the code we have. When I thought about how to implement it, I reversed my opinion as this makes the example code even longer and more complicated. 

I think I am still in favour of keeping the two examples but to change one of them to show case more of ODLs tools. We could also rename them to "pdhg_linear_convergence_rate" and "pdhg_primal_1k2_rate" to highlight why both of them are actually there. In any case, I would like to settle this sooner than later. It keeps dragging on and on.

Reply:
I don't mind having both examples as such. My worries are mainly that the `examples/solvers` folder becomes really crowded and hard to navigate. We have a top-level README to help (which reminds me that these examples should be mentioned there 😄), but still.

A way of organizing things better would be to keep the "basic" examples where they are now, in `examples/solvers`, and to create a subfolder `advanced` or so for stuff like this.

Reply:
I would like an advanced examples folder. However, I feel that this discussion has been there a few times. Thus, I propose to merge things the way they are and to open a new issue with this.

Reply:
Could be. You're right, let's not keep this from going in any longer.

The fact that this seems to be a recurring discussion indicates that it lacks a place for discussion. Would you mind opening an issue on that, i.e., finding a place for advanced examples?

Reply:
Done, see #1209